### PR TITLE
meson: fix build with case-sensitive store on Darwin

### PR DESCRIPTION
**Pull request by reckenrode**
###### Description of changes

I run Darwin with a case-sensitive store because it’s offered as an option. Meson is the first thing that actually broke because of it. This patch corrects the casing of “foundation” references in some of the tests that expect to find it.

I did not run `nixpkgs-review` because this change causes a lot of rebuilds, but I did rebuild a number of packages as part of my work on #229786.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Two basic questions:

1. The case-insensitve store use case will still be functioning?
2. Shouldn't this patch be sent to Meson developers, too?

As I have said before, I have no access to a Darwin machine.

**Comment by reckenrode** [comment]
The patch shouldn’t affect users on case-insensitive stores. All it does is change the casing of `foundation` to `Foundation` on a couple of tests. I have submitted a patch upstream as mesonbuild/meson#11820. I updated my PR here to use `fetchpatch` instead of including the patch in nixpkgs.

File: pkgs/development/tools/build-managers/meson/default.nix [hunk 1]
```diff
@@ -26,6 +26,12 @@ python3.pkgs.buildPythonApplication rec {
   };
 
   patches = [
+    # Fix Meson tests that fail when the Nix store is case-sensitive APFS.
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/mesonbuild/meson/pull/11820.patch";
```
Comment (wegank) [inline pkgs/development/tools/build-managers/meson/default.nix:31]:
```suggestion
      url = "https://github.com/mesonbuild/meson/pull/11820.patch";
```

Comment (reckenrode) [reply pkgs/development/tools/build-managers/meson/default.nix]: Thanks, I fixed the URL and pushed an updated commit.

File: pkgs/development/tools/build-managers/meson/default.nix [hunk 2]
```diff
@@ -26,6 +26,12 @@ python3.pkgs.buildPythonApplication rec {
   };
 
   patches = [
+    # Fix Meson tests that fail when the Nix store is case-sensitive APFS.
+    (fetchpatch {
+      url = "ttps://github.com/mesonbuild/meson/pull/11820.patch";
```
Comment (wegank) [inline pkgs/development/tools/build-managers/meson/default.nix:31]:
```suggestion
      url = "https://github.com/mesonbuild/meson/pull/11820.patch";
```

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
@ofborg build meson meson.passthru.tests

**Comment by eli-schwartz** [comment]
> Shouldn't this patch be sent to Meson developers, too?

It is now merged upstream.

**Pull request merged**